coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -393,90 +383,117 @@ static boolean isUnionRecordSchemaCompatible(@Nonnull Schema leftSchema, @Nonnul
       boolean strictMode) {
     Preconditions.checkNotNull(leftSchema);
     Preconditions.checkNotNull(rightSchema);
 
-    switch (leftSchema.getType()) {
-      case BOOLEAN:
-      case BYTES:
-      case DOUBLE:
-      case FLOAT:
-      case INT:
-      case LONG:
-      case STRING:
-      case NULL:
-        return leftSchema.getType() == rightSchema.getType();
-      case FIXED:
-        boolean isSameType = leftSchema.getType() == rightSchema.getType();
-        boolean isSameNamespace = Objects.equals(leftSchema.getNamespace(), rightSchema.getNamespace());
-
-        return isSameType && (!strictMode || isSameNamespace);
-      case ENUM:
-        boolean isSameEnumType = (leftSchema.getType() == rightSchema.getType());
-        boolean isSameSymbolSize = (leftSchema.getEnumSymbols().size() == rightSchema.getEnumSymbols().size());
-        boolean isSameEnumNamespace = Objects.equals(leftSchema.getNamespace(), rightSchema.getNamespace());
-
-        return isSameEnumType && isSameSymbolSize && (!strictMode || isSameEnumNamespace);
-      case RECORD:
-        return leftSchema.getType() == rightSchema.getType()
-            && isUnionRecordSchemaCompatible(leftSchema, rightSchema, strictMode);
-
-      case MAP:
-        return leftSchema.getType() == rightSchema.getType()
-            && isUnionSchemaCompatible(leftSchema.getValueType(), rightSchema.getValueType(), strictMode);
-
-      case ARRAY:
-        return leftSchema.getType() == rightSchema.getType()
-            && isUnionSchemaCompatible(leftSchema.getElementType(), rightSchema.getElementType(), strictMode);
+    Schema.Type leftSchemaType = leftSchema.getType();
+    Schema.Type rightSchemaType = rightSchema.getType();
+    if (leftSchemaType == NULL) {
+      return makeNullable(rightSchema);
+    }
+    if (rightSchemaType == NULL) {
+      return makeNullable(leftSchema);
+    }
+    if (isNullableType(leftSchema) || isNullableType(rightSchema)) {
+      return makeNullable(getUnionFieldSchema(makeNonNullable(leftSchema), makeNonNullable(rightSchema), strictMode));
+    }
 
-      case UNION:
-        boolean isSameUnionType = (leftSchema.getType() == rightSchema.getType());
-        boolean isBothNullableType =
-            AvroSerdeUtils.isNullableType(leftSchema) && AvroSerdeUtils.isNullableType(rightSchema);
+    if (leftSchemaType == rightSchemaType) {
+      Schema outputSchema;
+      boolean compatible = true;
 
-        Schema leftOtherType = AvroSerdeUtils.getOtherTypeFromNullableType(leftSchema);
-        Schema rightOtherType = AvroSerdeUtils.getOtherTypeFromNullableType(rightSchema);
-        boolean isOtherTypeUnionCompatible = isUnionSchemaCompatible(leftOtherType, rightOtherType, strictMode);
+      // Each of the following case options adjusts the value of "compatible" with additional conditions
+      // and sets the value of "outputSchema". This method will return "outputSchema" if "compatible" is true.
```

Comment:
`outputSchema` and `compatible` can be removed

Reply:
Done.